docs/source/buffy.rst

```diff
@@ -1,5 +1,145 @@
+shape of the **TWF** expected for each type of sensor: ``pmtrd`` and ``sipmtrd`` (PMTs and SiPMs respectively) based on the *bufferisation* parameters provided. Pictures below represent the output for PMTs and SiPMs waveforms.
+
+.. image:: images/buffy/pmt_wft.png
+  :width: 350
+.. image:: images/buffy/sipm_wft.png
+  :width: 350
+
+
+This process is separated in the following tasks in the city:
+
+• :ref:`Histogram creation <Histogram>`
+• :ref:`Signal Search <Signal-Search>`
+• :ref:`Synchronisation and trigger separation <Trigg-Separation>`
+
+Buffy output also includes ``/Run/runInfo`` and ``/Run/events`` tables as the ones generated during data taking.
+
+.. note::
+  Historically, Buffy is based in an initial code of detsim (https://github.com/next-exp/IC/tree/master/invisible_cities/detsim) and most of its functions are located in that path but they are independent to :doc:`detsim` city.
+
+
+.. _Histogram:
+
+Histogram creation
+::::::::::::::::::
+
+As it was highlighted earlier, NEXUS information about sensor hits (``/MC/sns_response``) comes binned in time based on when a sensor sees some energy deposition.
+This means that the ``time_bin`` column numbers are increasing for a given event, but they can have gaps since empty time bins are not stored. This initial part of the city
+checks the time stamp of an event according to the sensors' response and defines histograms of charge distribution between [:math:`t_{min}`, :math:`t_{max}`], being:
+
+• :math:`t_{min}`: the time stamp of the first charge deposition of the event,
+• :math:`t_{max}`: defined considering that ``max_time`` =  :math:`t_{max}` - :math:`t_{min}`.
+
+.. image:: images/buffy/histogram_creation.png
+  :width: 800
+
+These histograms (one for PMTs and another for SiPMs) are defined by summing all individual sensors. This step restores also empty bins by padding zeros in between separate signals, and sample
+the histograms according to the binning of each type of sensor (``pmt_width`` and ``sipm_width``). Sampling widths are included in the simulation parameters (``/MC/info``), and depend on the type of sensor and detector.
+Normally correspond to 25 :math:`ns` for PMTs and and 1 :math:`\mu s` for SiPMs.
+
+.. _Signal-Search:
+
+Signal Search
+::::::::::::::::::
+
+Once the charge is distributed in the previously defined histograms, the code searches for signal-like events.
+It takes the PMT sum histogram and looks for the first value of the binned charge above a certain threshold (``trigger_threshold``), and defines the trigger time, :math:`t_{trigger}`.
+Waveforms are therefore defined for PMTs:
+
+• shifting the times of the charge histogram such that the first value over threshold (:math:`t_{trigger}`) falls at the time defined as ``pre_trigger``;
+• considering a specific length based on ``buffer_length``/``pmt_width``.
```

Comment:
I think this sentence should be more concrete:
- set the length (in number of samples) as requested in the config parameters (`buffer_length/pmt_width`)

Reply:
Done